src/rendering/SceneManager.cc

```diff
@@ -702,23 +702,52 @@ rendering::GeometryPtr SceneManager::LoadGeometry(const sdf::Geometry &_geom,
   }
   else if (_geom.Type() == sdf::GeometryType::MESH)
   {
-    auto fullPath = asFullPath(_geom.MeshShape()->Uri(),
-        _geom.MeshShape()->FilePath());
-    if (fullPath.empty())
+    common::MeshManager *meshManager =
+        common::MeshManager::Instance();
+    std::string meshUri;
+    rendering::MeshDescriptor descriptor;
+    if (meshManager->IsValidFilename(_geom.MeshShape()->Uri()))
     {
-      gzerr << "Mesh geometry missing uri" << std::endl;
+      // Assume absolute path to mesh file
+      auto fullPath = asFullPath(_geom.MeshShape()->Uri(),
+          _geom.MeshShape()->FilePath());
+      if (fullPath.empty())
+      {
+        gzwarn << "Failed to load mesh from [" << fullPath
+               << "]." << std::endl;
+        return geom;
+      }
+      meshUri = fullPath;
+      descriptor.mesh = meshManager->Load(meshUri);
+    }
+    else if (common::URI(_geom.MeshShape()->Uri()).Scheme() == "name")
```

Comment:
Should this be checked first? If we have `name://foo.dae`, but there's a `foo.dae` somewhere in the file search path, what would be the behavior?

Reply:
ok swapped order of check so that the scheme determines how the mesh will be loaded. So ror `name://foo.dae` it'll assume it's an in-memory mesh.